src/core/backbone_router/backbone_tmf.cpp

```diff
@@ -0,0 +1,75 @@
+ *  modification, are permitted provided that the following conditions are met:
+ *  1. Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *  2. Redistributions in binary form must reproduce the above copyright
+ *     notice, this list of conditions and the following disclaimer in the
+ *     documentation and/or other materials provided with the distribution.
+ *  3. Neither the name of the copyright holder nor the
+ *     names of its contributors may be used to endorse or promote products
+ *     derived from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ *  AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ *  IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ *  ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ *  LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ *  CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ *  SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ *  INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ *  CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ *  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/**
+ * @file
+ *   This file implements Backbone Thread Management Framework (TMF) functionalities.
+ */
+
+#include "backbone_tmf.hpp"
+
+#if OPENTHREAD_FTD && OPENTHREAD_CONFIG_BACKBONE_ROUTER_ENABLE
+
+#include "common/locator-getters.hpp"
+
+namespace ot {
+namespace BackboneRouter {
+
+otError BackboneTmfAgent::Start(void)
+{
+    return Coap::Start(kBackboneUdpPort, OT_NETIF_BACKBONE);
+}
+
+otError BackboneTmfAgent::Filter(const ot::Coap::Message &aMessage,
+                                 const Ip6::MessageInfo & aMessageInfo,
+                                 void *                   aContext)
+{
+    OT_UNUSED_VARIABLE(aMessage);
+
+    return static_cast<BackboneTmfAgent *>(aContext)->IsBackboneTmfMessage(aMessageInfo) ? OT_ERROR_NONE
+                                                                                         : OT_ERROR_NOT_TMF;
+}
+
+bool BackboneTmfAgent::IsBackboneTmfMessage(const Ip6::MessageInfo &aMessageInfo)
+{
+    const Ip6::Address &dst = aMessageInfo.GetSockAddr();
+    const Ip6::Address &src = aMessageInfo.GetPeerAddr();
+
+    // A Backbone TMF message must comply with following rules:
+    // The destination must be one of:
+    //     1. All Network BBRs (Link-Local scope)
+    //     2. All Domain BBRs (Link-Local scope)
+    //     3. A Backbone Link-Local address
+    // The source must be a Backbone Link-local address.
+    return (dst.IsLinkLocal() || dst.IsLinkLocalMulticast()) && src.IsLinkLocal();
```

Comment:
I was thinking about this too, but thought it might not matter that much. 
 
Note that if the code here runs, the destination of the message **MUST** be a subscribed Multicast Address on the Backbone interface. The other possible Multicast Addresses are `Solicited-node Multicast Addresses` which are also link-local. 
We should never receive TMF messages with `Solicited-node Multicast Addresses` as destinations. But even if we did receive a message like that, it does not harm to process it? 

Thoughts?

Reply:
If it doesn't cost much to be accurate, why leave it vague?
I may prefer a bit to update the code to be consistent with the rules `1.`, `2.` you defined in the comments.

Reply:
revised accordingly.